### ISSUE TITLE
Document how to run tests

### ISSUE THREAD

**Issue by BlueDrink9** [autokey/autokey#327]
Enhancement

## Description

I want to contribute, but cannot get the test cases to pass.

I've tried running `python -m unittest discover --top-level-directory . -start-directory test --pattern *test.py`

The tests are unable to import the relevant files, probably because there is an `autokey` subdir under the `lib` dir they are trying to import from.

Please document how to run the test cases to allow easier contribution. A `CONTRIBUTERS.MD` would be good as well.

**Comment by luziferius** [comment]
Thank you for taking interest in AutoKey! 
TLDR: `git checkout develop; python3 setup.py test`

The current tests are deprecated and won’t work. They are still on Python 2 and nobody cared to update them throughout the years, not even the original developer. They test things long gone in the project code… The old tests are scheduled to be replaced by a new suite I wrote for the next major release, which will be 0.96.0. (Whenever that happens…)

The current situation is a bit unfortunate: The new suite lives in the `develop` branch, which accumulates new features, but I can’t backport them to `master` without introducing merge conflicts later on. So until `develop` gets merged in, you’d have to switch to `develop` to run the tests. 

(It requires `py.test` and `PyHamcrest` as new test-time only dependencies.)
Running the tests is simple: Checkout `develop` and run `python3 setup.py test`

When you want to contribute new features or fix things, you are free to take virtually any task you wish. Just open a PR for discussion and I’ll try to answer any questions that arise. I suggest writing new features on top of `develop`.

**Comment by BlueDrink9** [comment]
Thank you!

**Comment by BlueDrink9** [comment]
As an aside, is there a roadmap for what it would take to release 0.96.0?
For contributers to help with?

**Comment by luziferius** [comment]
You can look at the milestones defined. E g. https://github.com/autokey/autokey/milestone/2

I don't think that solving all is required for the release. It accumulated a bit.

Just comment in one of the issues you want to claim.

Edit: Unrelated side note: The mobile GitHub UI is flimsy and does unwanted things...
